bin/run_ci_dataset.sh

```diff
@@ -39,8 +39,8 @@ usage() {
     print_error
     print_error "Specific options:"
     print_error "   -d          Dataset name"
-    print_error "   -g          Middleware generation number (int)"
-    print_error "   -p          Pipeline to run (Gen 3 only)"
+    print_error "   -g          Middleware generation number (int) [currently unused]"
```

Comment:
Nothing's been missed -- the Jenkins script [still calls this argument](https://github.com/lsst-dm/jenkins-dm-jobs/blob/1aa272fe6cda2e557831dc1e0245cdb424ee8e06/pipelines/scipipe/ap_verify.groovy#L429).

The main reasons I don't want to remove it completely are

a) this would require unraveling a great deal of Jenkins code, all the way down to the config format and everything that depends on *that*
b) a lot of how to support multiple Butler generations on the Jenkins side took a long time to work out, and I'd rather keep it around (despite the triviality of `gen === 3`) than have to reinvent it when we have a similar problem.